### PR TITLE
feat!: async support on message type and schema registry resolvers

### LINKED ISSUE #302: Async support on ISchemaRegistryTypeNameResolver
Review interface and implementation to use Async/Await.

From
```csharp
    public interface ISchemaRegistryTypeNameResolver
    {
        string Resolve(int schemaId);
    }
```

To
```csharp
    public interface ISchemaRegistryTypeNameResolver
    {
        Task<string> ResolveAsync(int schemaId);
    }
```

### PR DESCRIPTION
**Pull request by jose-sousa-8**
# Description

- Adds async support for both **Consume** and **Produce** message type resolver. 
- Adds async support for the **Resolve** method in the schema registry type resolver.

Fixes #302  #305 

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have added tests to cover my changes
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/KafkaFlow.Serializer/IMessageTypeResolver.cs [hunk 1]
```diff
@@ -12,12 +13,12 @@ public interface IMessageTypeResolver
...
         /// <returns></returns>
-        Type OnConsume(IMessageContext context);
+        Task<Type> OnConsumeAsync(IMessageContext context);
 
         /// <summary>
         /// Stores the message type somewhere when producing
         /// </summary>
         /// <param name="context">The <see cref="IMessageContext"/> containing the message and the metadata</param>
-        void OnProduce(IMessageContext context);
+        Task OnProduceAsync(IMessageContext context);
```
Comment (filipeesch) [inline src/KafkaFlow.Serializer/IMessageTypeResolver.cs:22]: Please use ValueTask as return type

File: src/KafkaFlow.Serializer/IMessageTypeResolver.cs [hunk 2]
```diff
@@ -12,12 +13,12 @@ public interface IMessageTypeResolver
         /// </summary>
         /// <param name="context">The <see cref="IMessageContext"/> containing the message and the metadata</param>
         /// <returns></returns>
-        Type OnConsume(IMessageContext context);
+        Task<Type> OnConsumeAsync(IMessageContext context);
```
Comment (filipeesch) [inline src/KafkaFlow.Serializer/IMessageTypeResolver.cs:16]: Please use ValueTask as return type

File: src/KafkaFlow.Serializer/IMessageTypeResolver.cs (hunk 1 above)
Comment (jose-sousa-8) [reply src/KafkaFlow.Serializer/IMessageTypeResolver.cs:22]: Is it because the implementations of this interface are synchronous?

File: src/KafkaFlow.Serializer.SchemaRegistry.ConfluentProtobuf/ConsumerConfigurationBuilderExtensions.cs [hunk 3]
```diff
@@ -1,8 +1,7 @@
-﻿namespace KafkaFlow
+﻿namespace KafkaFlow.Serializer.SchemaRegistry
```
Comment (filipeesch) [inline src/KafkaFlow.Serializer.SchemaRegistry.ConfluentProtobuf/ConsumerConfigurationBuilderExtensions.cs:1]: please undo this change. The configuration builders should be in KafkaFlow root namespace to make it easier for the client application to use

File: src/KafkaFlow.Serializer.SchemaRegistry.ConfluentProtobuf/ProducerConfigurationBuilderExtensions.cs [hunk 4]
```diff
@@ -1,9 +1,8 @@
-﻿namespace KafkaFlow
+﻿namespace KafkaFlow.Serializer.SchemaRegistry
```
Comment (filipeesch) [inline src/KafkaFlow.Serializer.SchemaRegistry.ConfluentProtobuf/ProducerConfigurationBuilderExtensions.cs:1]: please undo this change. The configuration builders should be in KafkaFlow root namespace to make it easier for the client application to use

**Review by filipeesch — CHANGES_REQUESTED** [review]

File: src/KafkaFlow.Serializer/IMessageTypeResolver.cs (hunk 1 above)
Comment (filipeesch) [reply src/KafkaFlow.Serializer/IMessageTypeResolver.cs]: This is to reduce the heap allocations. The "hot path" of this interface implementation is sync, so it's better to use a ValueTask in these cases.

**Review by filipeesch — APPROVED** [review]

**Review by joelfoliveira — APPROVED** [review]

**Review by brmagadutra — APPROVED** [review]
lgmt

**Review by erik-catalao — APPROVED** [review]
LGTM

**Review by JoaoRodriguesGithub — APPROVED** [review]
